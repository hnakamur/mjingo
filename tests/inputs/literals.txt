{}
---
{{ true }}
{{ false }}
{{ null }}
{{ 1.0 }}
{{ 2 }}
{{ [1, 2, 3] }}
{{ {"foo": "bar"} }}
{{ -9223372036854775808 }}
{{ -170141183460469231731687303715884105728 }}
{{ 170141183460469231731687303715884105727 }}
{{ 340282366920938463463374607431768211455 }}
