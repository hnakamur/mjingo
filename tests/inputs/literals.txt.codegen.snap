(
    [
        00000 | EmitRaw("{}\n---\n")  [line 1],
        00001 | LoadConst(true)  [line 3],
        00002 | Emit,
        00003 | EmitRaw("\n"),
        00004 | LoadConst(false)  [line 4],
        00005 | Emit,
        00006 | EmitRaw("\n"),
        00007 | Lookup("null")  [line 5],
        00008 | Emit,
        00009 | EmitRaw("\n"),
        00010 | LoadConst(1.0)  [line 6],
        00011 | Emit,
        00012 | EmitRaw("\n"),
        00013 | LoadConst(2)  [line 7],
        00014 | Emit,
        00015 | EmitRaw("\n"),
        00016 | LoadConst([1, 2, 3])  [line 8],
        00017 | Emit,
        00018 | EmitRaw("\n"),
        00019 | LoadConst({"foo": "bar"})  [line 9],
        00020 | Emit,
        00021 | EmitRaw("\n"),
        00022 | LoadConst(-9223372036854775808)  [line 10],
        00023 | Emit,
        00024 | EmitRaw("\n"),
        00025 | LoadConst(170141183460469231731687303715884105728)  [line 11],
        00026 | Emit,
        00027 | EmitRaw("\n"),
        00028 | LoadConst(170141183460469231731687303715884105727)  [line 12],
        00029 | Emit,
        00030 | EmitRaw("\n"),
        00031 | LoadConst(340282366920938463463374607431768211455)  [line 13],
        00032 | Emit,
    ],
    {},
)
